src/main/java/org/mule/extension/http/internal/request/HttpRequestFactory.java

```diff
@@ -168,6 +161,27 @@ public HttpRequest create(HttpRequesterConfig config, String uri, String method,
     return builder.build();
   }
 
+  private void addCookiesHeader(HttpRequesterConfig config, String uri, HttpRequestBuilder builder) {
+    try {
+      List<String> cookies = config.getCookieManager().get(builder.getUri(), emptyMap()).get(COOKIE);
+      if (cookies != null && cookies.size() > 0) {
+        // The RFC-6265, section 5.4 says:
+        //   If the user agent does attach a Cookie header field to an HTTP
+        //   request, the user agent MUST send the cookie-string (defined below)
+        //   as the value of the header field.
+        //
+        // So we should concatenate the cookies
+        StringBuilder sb = new StringBuilder(cookies.get(0));
```

Comment:
try to presize this according to how many headers will be added. The default size is too small and it may grow many times when there are many headers, hurting performance